### PR TITLE
New test examples and more detailed information

### PR DESCRIPTION
**Pull request by De1ik**
New example of integration tests:
- Rust Anchor Testing Environment
- @solana/web3.js

More detailed explanation for type of tests and comparison of different tools with the information of benefits and disadvantages.

Initialize instruction was changed to include user.key into the seeds to avoid the error like "data already exist on chain" when run all tests at once.

### CODE REVIEW

File: 5.lesson/README.md [hunk 1]
```diff
@@ -10,6 +10,14 @@ This week, we are taking a look at a few common errors you’re likely to run in
...
   - [Units Tests](#units-tests)
   - [Integration Tests](#integration-tests)
+  - [Popular Tools for Integration Testing in Rust](#popular-tools-for-integration-testing-in-rust)
+  - [Popular Tools for Integration Testing in Solana Client](#popular-tools-for-integration-testing-in-solana-client)
+  - [How to Choose Between These Tools](#how-to-choose-between-these-tools)
+- [Testing Solana Programs](#testing-solana-programs)
+  - [Why Testing Matters](#why-testing-matters)
+  - [Tooling Overview](#tooling-overview)
+  - [Comparison](#comparison)
+  - [Examples](#examples)
```
Comment (hyckomatej) [inline 5.lesson/README.md]: some toc links dont work

File: 5.lesson/README.md [hunk 2]
```diff
@@ -23,13 +31,59 @@ This week, we are taking a look at a few common errors you’re likely to run in
 ## Types of Tests
 You should always aim for high test coverage. This includes testing both the "happy path" (successful cases) and "unhappy path" (error cases).
 
+> [!TIP]
+> Examples of code and guide how to run each test you can find inside [test-examples](./test-examples/) folder.
```
Comment (hyckomatej) [inline 5.lesson/README.md]: sounds very unnatural, rephrase please

File: 5.lesson/README.md [hunk 3]
```diff
@@ -43,17 +97,107 @@ mod tests {
...
+
+The Solana ecosystem provides several frameworks designed to facilitate integration testing, each offering a different balance between control, abstraction, and developer convenience. The most notable among them are solana-program-test, Anchor’s testing framework, and client-based testing libraries such as Solana Web3.js or Solana Kit.
+
+### Popular Tools for Integration Testing in Rust
+
+#### Solana Program Test Framework
+
+The solana-program-test crate is the backbone of integration testing in Solana’s Rust ecosystem. It launches a lightweight, in-process version of the Solana runtime that behaves nearly identically to a local validator but runs fully in memory. Developers can register one or more programs, create test accounts, send instructions, and inspect the resulting on-chain state — all without deploying to an external cluster.
```
Comment (hyckomatej) [inline 5.lesson/README.md]: sometimes you use backticks `` to highlight various crates/terms but other times not. try to keep the style consistent.  eg solana-program-test

File: 5.lesson/README.md [hunk 4]
```diff
@@ -43,17 +97,107 @@ mod tests {
...
+
+This framework is good for testing interactions between multiple accounts and verifying instruction-level behavior. For example, a developer might simulate the full lifecycle of a voting process, from poll initialization to casting votes and closing results, while asserting that all state transitions occur correctly. The framework also supports testing multiple programs simultaneously, which is useful for projects involving CPIs or modular on-chain architectures. 
+
+#### Anchor Testing Environment
+
+For projects built with Anchor, integration testing becomes even more streamlined. Anchor internally uses `solana-program-test`, but enhances it with an expressive and ergonomic API. Instead of manually constructing transactions and serializing account data, developers can interact with their programs directly using method calls, with account validation and PDA derivation handled automatically based on the program’s IDL.
```
Comment (hyckomatej) [inline 5.lesson/README.md:117]: see here solana-program-test is nicely highlighted

File: 5.lesson/README.md [hunk 5]
```diff
@@ -80,17 +224,73 @@ try {
...
+
+- If you need to have rust integration tests and your on-chain program is built with Anchor you will probably use Anchor Testing Environment, otherwice Solana Program Test Framework.
+
+- If your on-chain program is built with Anchor and you value developer productivity and strong typing, start with `@coral-xyz/anchor`. It gives you the fastest path to writing tests aligned with your program’s IDL and account structure.
+
+- If you are working with a custom on-chain Rust program (not using Anchor), or you require custom instruction layout or CPIs across non-Anchor programs, you’ll likely opt for `@solana/web3.js`, accepting the extra boilerplate for the sake of flexibility.
+
+- If your testing scope covers client application flows (wallet interactions, UI simulation, batching, and front-end logic) and you prefer a library optimized for developer experience, you may explore gill (or Kit) — but be aware of less matured community examples for strict integration tests of on-chain programs.
+
+In practice, many teams adopt a layered approach: using Anchor’s test tooling for most integration tests, occasionally falling back to web3.js for the “weird corner cases”.
```
Comment (hyckomatej) [inline 5.lesson/README.md]: no need to include comments like this, seems like some BS embelishment added by llm. we want the learning materials to be information dense while being easy to understand. avoid unnecessary comments like this.

File: 5.lesson/test-examples/README.md [hunk 6]
```diff
@@ -0,0 +1,84 @@
...
+├── anchor-test.ts             # Anchor client test (Mocha/Chai)
+└── web3-raw-test.ts           # Raw web3.js instruction test
+```
+## 🚀 Running Tests
+
+- Rust unit + integration (integration test requires active validator, read next section for more info): `cd programs/test-examples && cargo test`
+- Client based tests: `anchor test`
+
+## ⚠️ Important Notes for Rust Integration Tests
+When you use `anchor test` you do not need to start validator by yourself, because anchor do it automatically:
```
Comment (hyckomatej) [inline 5.lesson/test-examples/README.md]: ...anchor does it..              check grammar using llm

**Review by hyckomatej — CHANGES_REQUESTED** [review]
I really like the new content and examples but form needs a bit of work. Left a few comments on shortcomings i noticed. 

Our aim is to deliver learning materials that are as information dense as possible while being relatively easy to understand to people that are already well versed in programming and tech. No need to use fancy terms or other extra fluff added by llms (there is no word count requirement). 

Go through the readmes, fix requested changes, have llm check the grammar, get rid of unnecessary filler words and sentences. Also make sure the style is consistent.

Overall, great work! Content is the most important, and you've already got that covered. Now you just need to slightly improve the form. Thanks!

**Review by hyckomatej — APPROVED** [review]
lgtm, thanks
